File: projects/03/grades.rb
# grades.rb is a driver function for letterGrade
# By: Zach Wibbenmeyer
# Date: February 26, 2016
#############################################################


# letterGrade()  method - returns a letter grade for the average given
# @param: average (type = integer
# @return: a grade (type = character)
# Precondition: the average entered must be an integer
def letterGrade(average)
	case average / 10
	when 9, 10
		'A'
	when 8
		'B'
	when 7
		'C'
	when 6
		'D'
	else
		'F'
	end
end

# Main program
if __FILE__ == $0
	
	# Prompt for the average
	print "Enter the average grade: "
	# Get the average
	avg = gets.chomp.to_i
	# Call the method to return the grade
	puts letterGrade(avg)
end

File: projects/03/grades.java
/* grades.java is a driver for function letterGrade().
 * By: Zach Wibbenmeyer
 * Date: February 26, 2016
 ****************************************************************/


// Import the necessary libraries
import java.util.Scanner;


public class grades {

   /* letterGrade() method - Displays the letter grade for the given average
    * @param: average (type = int)
    * @return: various letter grades (type = char)
    * Precondition: An int for the average must be entered
    */
   public static char letterGrade(int average) {
      // Calculates the average
      switch (average / 10) {
	 // Various cases
         case 10: case 9:
		    return 'A';
		 case 8:
		    return 'B';
		 case 7:
		    return 'C';
		 case 6:
		    return 'D';
		 default:
		    return 'F';
	  }
   }


   // Main program
   public static void main(String[] args) {
      // Prompt for the average
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter the average score: ");

      // Read the average
      int avg = keyboard.nextInt();
      // Print the letter grade
      System.out.println(letterGrade(avg));
   }
}
